e2e/src/authentication/login.e2e-spec.ts

```diff
@@ -0,0 +1,80 @@
+import { LoginPage } from './login.po';
+import { browser, logging, until } from 'protractor';
+
+describe('Login', () => {
+  let page: LoginPage;
+
+  beforeEach(() => {
+    page = new LoginPage();
+  });
+
+  describe('Given valid credential', () => {
+    it('enables form submit button', async () => {
+      await page.navigateTo();
+      await page.fillEmail('email@nimblehq.co');
+      await page.fillPassword('password');
+
+      expect(await (await page.getSubmitButton()).getAttribute('disabled')).toBeFalsy();
+    });
+
+    it('sets access token to local storage', async () => {
+      await page.navigateTo();
+      await page.LoginWith('dev@nimblehq.co', '12345678');
+
+      expect(localStorage.getItem('USER_ACCESS_TOKEN')).not.toBeNull();
+      expect(localStorage.getItem('USER_TOKEN_TOKEN')).not.toBeNull();
+    });
+
+    it('redirects to root url when hit login button', async () => {
+      await page.navigateTo();
+      await page.LoginWith('dev@nimblehq.co', '12345678');
+
+      browser.waitForAngular();
+      browser.wait(() => {
+        return until.urlIs('/');
+      }, 2000);
```

Comment:
I see this timeout value of `2000` a lot. How about changing the value globally in `jasmineNodeOpts` so that it can be avoided?

https://github.com/carryall/survey-challenge/blob/34c19b301e3736f4447064d2da24533e0c431bf5/e2e/protractor.conf.js#L25-L29

So it can be omitted. Maybe 30 seconds might be too much but I doubt it will take that long to be completed in any case.